server/src/internalClusterTest/java/org/opensearch/remotemigration/DocrepToRemoteDualReplicationIT.java

```diff
@@ -0,0 +1,420 @@
+
+@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.TEST, numDataNodes = 0, autoManageMasterNodes = false)
+public class DocrepToRemoteDualReplicationIT extends MigrationBaseTestCase {
+    private String REMOTE_PRI_DOCREP_REP = "remote-primary-docrep-replica";
+    private String REMOTE_PRI_DOCREP_REMOTE_REP = "remote-primary-docrep-remote-replica";
+    private String FAILOVER_REMOTE_TO_DOCREP = "failover-remote-to-docrep";
+
+    /*
+    Scenario:
+    - Starts 2 docrep backed node
+    - Creates index with 1 replica
+    - Index some docs
+    - Start 1 remote backed node
+    - Move primary copy from docrep to remote through _cluster/reroute
+    - Index some more docs
+    - Assert primary-replica consistency
+     */
+    public void testRemotePrimaryDocRepReplica() throws Exception {
+        internalCluster().setBootstrapClusterManagerNodeIndex(0);
+        internalCluster().startClusterManagerOnlyNode();
+
+        logger.info("---> Starting 2 docrep data nodes");
+        internalCluster().startDataOnlyNode();
+        internalCluster().startDataOnlyNode();
+        internalCluster().validateClusterFormed();
+        assertEquals(internalCluster().client().admin().cluster().prepareGetRepositories().get().repositories().size(), 0);
+
+        logger.info("---> Creating index with 0 replica");
+        Settings oneReplica = Settings.builder().put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 1).build();
+        createIndex(REMOTE_PRI_DOCREP_REP, oneReplica);
+        ensureGreen(REMOTE_PRI_DOCREP_REP);
+
+        int initialBatch = randomIntBetween(1, 1000);
+        logger.info("---> Indexing {} docs", initialBatch);
+        SyncIndexingService indexingService = new SyncIndexingService(REMOTE_PRI_DOCREP_REP, initialBatch);
+        indexingService.startIndexing();
+
+        initDocRepToRemoteMigration();
+
+        logger.info("---> Starting 1 remote enabled data node");
+        addRemote = true;
+        String remoteNodeName = internalCluster().startDataOnlyNode();
+        internalCluster().validateClusterFormed();
+        assertEquals(
+            internalCluster().client()
+                .admin()
+                .cluster()
+                .prepareGetRepositories(REPOSITORY_NAME, REPOSITORY_2_NAME)
+                .get()
+                .repositories()
+                .size(),
+            2
+        );
+
+        String primaryShardHostingNode = primaryNodeName(REMOTE_PRI_DOCREP_REP);
+        logger.info("---> Moving primary copy from {} to remote enabled node {}", primaryShardHostingNode, remoteNodeName);
+        assertAcked(
+            internalCluster().client()
+                .admin()
+                .cluster()
+                .prepareReroute()
+                .add(new MoveAllocationCommand(REMOTE_PRI_DOCREP_REP, 0, primaryShardHostingNode, remoteNodeName))
+                .get()
+        );
+        ensureGreen(REMOTE_PRI_DOCREP_REP);
+
```

Comment:
Lets add an assertion on the moved shard's location. We can assert that the shard now resides on the remote node.